### ISSUE TITLE
@sveltejs/adapter-node@1.2.3 make memory leak

### ISSUE THREAD

**Issue by stalkerg** [sveltejs/kit#9512]
### Describe the bug

Basically, every 5min, I have an extra 100mb to the process. Originally I thought about `undici` lib same as last time #8239 but seems like now times it's something new. 

### Reproduction

I have a load on the production server for the node project. If it's will not an obvious reason I will try to make a reproduction. 
But seems like will be simpler to bisect changes. 

### System Info

```Shell
System:
    OS: Linux 5.4 Ubuntu 20.04.5 LTS (Focal Fossa)
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
    Memory: 36.95 GB / 62.81 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 18.12.1 - /usr/bin/node
    npm: 8.19.2 - /usr/bin/npm
  npmPackages:
    @sveltejs/adapter-node: 1.2.2 => 1.2.2 
    @sveltejs/kit: 1.14.0 => 1.14.0 
    @sveltejs/svelte-virtual-list: github:sveltejs/svelte-virtual-list => 3.0.1 
    svelte: 3.57.0 => 3.57.0 
    vite: ^4.2.1 => 4.2.1
```

### Severity

blocking an upgrade

**Comment by stalkerg** [comment]
Yes, it's still `undici`, adapter-node just merge such lib into `shim.js`. 
https://github.com/nodejs/undici/issues/1823 still valid.

**Comment by machycek** [comment]
I can vouch for this as well. We have a lot of restarts in our `adapter-node` app after upgrading. 
<img width="169" alt="image" src="https://user-images.githubusercontent.com/8449312/229387488-d300c700-5058-4b05-86dc-534d5cf0de41.png">

**Comment by bluwy** [comment]
Made a PR to revert this: #9591

Another way to workaround this for now is to use `overrides` in `package.json`

npm:
```json
"overrides": {
  "undici": "5.20.0"
}
```

**Comment by stalkerg** [comment]
@bluwy 
> Another way to workaround this for now is to use overrides in package.json

no, it's not working because undici merged into shims.js from `@sveltejs/adapter-node`

**Comment by bluwy** [comment]
I don't understand what you mean by undici merging into it? It's derived from [this file](https://github.com/sveltejs/kit/blob/6b6f4565c9f1486cb43e921113f145e001078090/packages/kit/src/exports/node/polyfills.js#L4), which is a dependency before you bundle your app.

**Comment by stalkerg** [comment]
@bluwy no, it's bundling during `@sveltejs/adapter-node` deploy to NPM; you can check the package and shim file here https://www.npmjs.com/package/@sveltejs/adapter-node?activeTab=code or in your node_modules folder.
